app/rest/rest/src/main/java/io/syndesis/rest/util/ReflectiveFilterer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.util;
+
+import io.syndesis.model.ListResult;
+import io.syndesis.rest.v1.util.PredicateFilter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class ReflectiveFilterer<T> implements Function<ListResult<T>, ListResult<T>> {
+
+    private final List<PredicateFilter<T>> predicateFilters;
+
+    public ReflectiveFilterer(Class<T> modelClass, List<FilterOptionsParser.Filter> filters) {
+        predicateFilters = new ArrayList<>(filters.size());
+        for (FilterOptionsParser.Filter f : filters) {
+            String op = f.getOperation().filter("="::equals).orElseThrow(
+                () -> new IllegalArgumentException(String.format("Unknown filter operation %s", f.getOperation()))
+            );
+
+            switch (op) {
+                case "=":
+                    String value = f.getValue().orElseThrow(
+                        () -> new IllegalArgumentException("Missing value in equality filter")
+                    );
+                    predicateFilters.add(equalityFilter(modelClass, f.getProperty(), value));
+                    break;
+                default:
+                    throw new IllegalArgumentException(String.format("Unknown filter operation %s", op));
```

Comment:
Should never reach here because of the safeguard above. But actually, I'd prefer to have the check here and remove the lambda exercise above.